diff --git a/node_modules/react-native/Libraries/Image/RCTImageLoader.mm b/node_modules/react-native/Libraries/Image/RCTImageLoader.mm
index 3571647..ad48dae 100644
--- a/node_modules/react-native/Libraries/Image/RCTImageLoader.mm
+++ b/node_modules/react-native/Libraries/Image/RCTImageLoader.mm
@@ -1040,6 +1040,7 @@ - (BOOL)canHandleRequest:(NSURLRequest *)request
 
 - (id)sendRequest:(NSURLRequest *)request withDelegate:(id<RCTURLRequestDelegate>)delegate
 {
+  @synchronized(self) {
   __block RCTImageLoaderCancellationBlock requestToken;
   requestToken = [self loadImageWithURLRequest:request callback:^(NSError *error, UIImage *image) {
     if (error) {
@@ -1069,6 +1070,7 @@ - (id)sendRequest:(NSURLRequest *)request withDelegate:(id<RCTURLRequestDelegate
 
   return requestToken;
 }
+}
 
 - (void)cancelRequest:(id)requestToken
 {
diff --git a/node_modules/react-native/Libraries/Image/RCTLocalAssetImageLoader.mm b/node_modules/react-native/Libraries/Image/RCTLocalAssetImageLoader.mm
index 8669ad3..6b9f81d 100644
--- a/node_modules/react-native/Libraries/Image/RCTLocalAssetImageLoader.mm
+++ b/node_modules/react-native/Libraries/Image/RCTLocalAssetImageLoader.mm
@@ -49,19 +49,27 @@ - (RCTImageLoaderCancellationBlock)loadImageForURL:(NSURL *)imageURL
                                  partialLoadHandler:(RCTImageLoaderPartialLoadBlock)partialLoadHandler
                                   completionHandler:(RCTImageLoaderCompletionBlock)completionHandler
 {
-  UIImage *image = RCTImageFromLocalAssetURL(imageURL);
-  if (image) {
-    if (progressHandler) {
-      progressHandler(1, 1);
+__block auto cancelled = std::make_shared<std::atomic<bool>>(false);
+  RCTExecuteOnMainQueue(^{
+    if (cancelled->load()) {
+      return;
     }
-    completionHandler(nil, image);
-  } else {
-    NSString *message = [NSString stringWithFormat:@"Could not find image %@", imageURL];
-    RCTLogWarn(@"%@", message);
-    completionHandler(RCTErrorWithMessage(message), nil);
-  }
+     UIImage *image = RCTImageFromLocalAssetURL(imageURL);
+    if (image) {
+      if (progressHandler) {
+        progressHandler(1, 1);
+      }
+      completionHandler(nil, image);
+    } else {
+      NSString *message = [NSString stringWithFormat:@"Could not find image %@", imageURL];
+      RCTLogWarn(@"%@", message);
+      completionHandler(RCTErrorWithMessage(message), nil);
+    }
+  });
 
-  return nil;
+  return ^{
+    cancelled->store(true);
+  };
 }
 
 @end
diff --git a/node_modules/react-native/scripts/.packager.env b/node_modules/react-native/scripts/.packager.env
new file mode 100644
index 0000000..361f5fb
--- /dev/null
+++ b/node_modules/react-native/scripts/.packager.env
@@ -0,0 +1 @@
+export RCT_METRO_PORT=8081
