diff --git a/node_modules/react-native-calendars/src/calendar-list/style.js b/node_modules/react-native-calendars/src/calendar-list/style.js
index 69a8e5b..5bb438c 100644
--- a/node_modules/react-native-calendars/src/calendar-list/style.js
+++ b/node_modules/react-native-calendars/src/calendar-list/style.js
@@ -20,8 +20,8 @@ export default function getStyle(theme={}) {
       color: appStyle.dayTextColor
     },
     calendar: {
-      paddingLeft: 15,
-      paddingRight: 15
+      // paddingLeft: 15,
+      // paddingRight: 15
     },
     staticHeader: {
       position: 'absolute', 
diff --git a/node_modules/react-native-calendars/src/calendar/day/basic/style.js b/node_modules/react-native-calendars/src/calendar/day/basic/style.js
index 14dae05..9c0ee20 100644
--- a/node_modules/react-native-calendars/src/calendar/day/basic/style.js
+++ b/node_modules/react-native-calendars/src/calendar/day/basic/style.js
@@ -9,10 +9,11 @@ export default function styleConstructor(theme={}) {
     base: {
       width: 32,
       height: 32,
-      alignItems: 'center'
+      alignItems: 'center',
+      justifyContent: 'center',
     },
     text: {
-      marginTop: Platform.OS === 'android' ? 4 : 6,
+      marginTop: 4,
       fontSize: appStyle.textDayFontSize,
       fontFamily: appStyle.textDayFontFamily,
       fontWeight: appStyle.textDayFontWeight,
diff --git a/node_modules/react-native-calendars/src/calendar/header/index.js b/node_modules/react-native-calendars/src/calendar/header/index.js
index dd77e90..6f7c882 100644
--- a/node_modules/react-native-calendars/src/calendar/header/index.js
+++ b/node_modules/react-native-calendars/src/calendar/header/index.js
@@ -212,7 +212,7 @@ class CalendarHeader extends Component {
           {rightArrow}
         </View>
         {!this.props.hideDayNames &&
-          <View style={this.style.week}>
+          <View style={[this.style.week, {justifyContent: 'space-between'}]}>
             {this.props.weekNumbers &&
               <Text allowFontScaling={false} style={this.style.dayHeader}></Text>
             }
diff --git a/node_modules/react-native-calendars/src/calendar/header/style.js b/node_modules/react-native-calendars/src/calendar/header/style.js
index 61cb68f..241a51d 100644
--- a/node_modules/react-native-calendars/src/calendar/header/style.js
+++ b/node_modules/react-native-calendars/src/calendar/header/style.js
@@ -43,7 +43,8 @@ export default function(theme={}) {
     week: {
       marginTop: 7,
       flexDirection: 'row',
-      justifyContent: 'space-around'
+      justifyContent: 'space-between',
+      paddingHorizontal: 16,
     },
     dayHeader: {
       marginTop: 2,
diff --git a/node_modules/react-native-calendars/src/calendar/style.js b/node_modules/react-native-calendars/src/calendar/style.js
index 358f5b4..94ec8de 100644
--- a/node_modules/react-native-calendars/src/calendar/style.js
+++ b/node_modules/react-native-calendars/src/calendar/style.js
@@ -7,8 +7,8 @@ export default function getStyle(theme={}) {
   const appStyle = {...defaultStyle, ...theme};
   return StyleSheet.create({
     container: {
-      paddingLeft: 5,
-      paddingRight: 5,
+      // paddingLeft: 5,
+      // paddingRight: 5,
       backgroundColor: appStyle.calendarBackground
     },
     monthView: {
@@ -18,7 +18,7 @@ export default function getStyle(theme={}) {
       marginTop: 7,
       marginBottom: 7,
       flexDirection: 'row',
-      justifyContent: 'space-around'
+      justifyContent: 'space-between',
     },
     ...(theme[STYLESHEET_ID] || {})
   });
diff --git a/node_modules/react-native-calendars/src/expandableCalendar/calendarProvider.js b/node_modules/react-native-calendars/src/expandableCalendar/calendarProvider.js
index aead2f7..6c9d6d4 100644
--- a/node_modules/react-native-calendars/src/expandableCalendar/calendarProvider.js
+++ b/node_modules/react-native-calendars/src/expandableCalendar/calendarProvider.js
@@ -1,6 +1,6 @@
 import _ from 'lodash';
 import React, {Component} from 'react';
-import {Animated, TouchableOpacity, View} from 'react-native';
+import {Animated, TouchableOpacity, View, Dimensions, Platform} from 'react-native';
 import PropTypes from 'prop-types';
 import XDate from 'xdate';
 
@@ -9,6 +9,7 @@ import {xdateToData} from '../interface';
 import styleConstructor from './style';
 import CalendarContext from './calendarContext';
 
+const screenHeight = Math.round(Dimensions.get('window').height);
 
 const commons = require('./commons');
 const UPDATE_SOURCES = commons.UPDATE_SOURCES;
@@ -36,6 +37,8 @@ class CalendarProvider extends Component {
     todayBottomMargin: PropTypes.number,
     /** Today button's style */
     todayButtonStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
+    /** Today button's text styl */
+    todayButtonTextStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
     /** The opacity for the disabled today button (0-1) */
     disabledOpacity: PropTypes.number
   }
@@ -47,7 +50,7 @@ class CalendarProvider extends Component {
     this.state = {
       date: this.props.date || XDate().toString('yyyy-MM-dd'),
       updateSource: UPDATE_SOURCES.CALENDAR_INIT,
-      buttonY: new Animated.Value(-props.todayBottomMargin || -TOP_POSITION),
+      buttonY: new Animated.Value((screenHeight + 50)),
       buttonIcon: this.getButtonIcon(props.date),
       disabled: false,
       opacity: new Animated.Value(1)
@@ -124,13 +127,15 @@ class CalendarProvider extends Component {
       const isToday = today === date;
 
       Animated.spring(this.state.buttonY, {
-        toValue: isToday ? TOP_POSITION : -this.props.todayBottomMargin || -TOP_POSITION,
+        // toValue: isToday ? TOP_POSITION : -this.props.todayBottomMargin || -TOP_POSITION,
+        toValue: isToday ? screenHeight + 50 : screenHeight - (Platform.OS === 'ios' ? 300 : 250),
         tension: 30,
         friction: 8,
         useNativeDriver: true
       }).start();
     }
   }
+;
 
   animateOpacity(disabled) {
     const {disabledOpacity} = this.props;
@@ -153,11 +158,12 @@ class CalendarProvider extends Component {
     const todayString = XDate.locales[XDate.defaultLocale].today || commons.todayString;
     const today = todayString.charAt(0).toUpperCase() + todayString.slice(1);
 
+
     return (
       <Animated.View style={[this.style.todayButtonContainer, {transform: [{translateY: buttonY}]}]}>
         <TouchableOpacity style={[this.style.todayButton, this.props.todayButtonStyle]} onPress={this.onTodayPress} disabled={disabled}>
           <Animated.Image style={[this.style.todayButtonImage, {opacity}]} source={buttonIcon}/>
-          <Animated.Text allowFontScaling={false} style={[this.style.todayButtonText, {opacity}]}>{today}</Animated.Text>
+          <Animated.Text allowFontScaling={false} style={[this.style.todayButtonText, this.props.todayButtonTextStyle, {opacity}]}>{today}</Animated.Text>
         </TouchableOpacity>
       </Animated.View>
     );
diff --git a/node_modules/react-native-calendars/src/expandableCalendar/style.js b/node_modules/react-native-calendars/src/expandableCalendar/style.js
index a226953..6d30bf1 100644
--- a/node_modules/react-native-calendars/src/expandableCalendar/style.js
+++ b/node_modules/react-native-calendars/src/expandableCalendar/style.js
@@ -73,7 +73,8 @@ export default function styleConstructor(theme = {}) {
     },
     weekDayNames: {
       flexDirection: 'row', 
-      justifyContent: 'space-between' 
+      justifyContent: 'space-between' ,
+      backgroundColor: 'red',
     },
     weekday: {
       width: 32,
@@ -89,10 +90,12 @@ export default function styleConstructor(theme = {}) {
     week: {
       marginTop: 7,
       marginBottom: 7,
-      paddingRight: 15,
-      paddingLeft: 15,
+      // paddingLeft: 15,
+      // paddingRight: 15,
       flexDirection: 'row',
-      justifyContent: 'space-around'
+      // justifyContent: 'space-around',
+      justifyContent: 'space-between',
+      // backgroundColor: 'blue'
     },
     dayHeader: {
       width: 32,
@@ -109,14 +112,14 @@ export default function styleConstructor(theme = {}) {
     todayButtonContainer: {
       alignItems: appStyle.todayButtonPosition === 'right' ? 'flex-end' : 'flex-start',
       position: 'absolute', 
-      left: 20, 
+      // left: 0, 
       right: 20, 
       bottom : 0
     },
     todayButton: {
-      height: commons.isTablet ? 40 : 28, 
+      height: commons.isTablet ? 40 : 34, 
       paddingHorizontal: commons.isTablet ? 20 : 12,
-      borderRadius: commons.isTablet ? 20 : 14,
+      borderRadius: commons.isTablet ? 20 : 18,
       flexDirection: appStyle.todayButtonPosition === 'right' ? 'row-reverse' : 'row',
       justifyContent: 'center',
       alignItems: 'center',
@@ -134,13 +137,13 @@ export default function styleConstructor(theme = {}) {
       })
     },
     todayButtonText: {
-      color: appStyle.todayButtonTextColor,
+      color: '#34353E',
       fontSize: commons.isTablet ? appStyle.todayButtonFontSize + 2 : appStyle.todayButtonFontSize, 
       fontWeight: appStyle.todayButtonFontWeight,
       fontFamily: appStyle.todayButtonFontFamily
     },
     todayButtonImage: {
-      tintColor: appStyle.todayButtonTextColor, 
+      tintColor: '#34353E', 
       marginLeft: appStyle.todayButtonPosition === 'right' ? 7 : undefined,
       marginRight: appStyle.todayButtonPosition === 'right' ? undefined : 7
     },
